### PR TITLE
feat(deployer): send deployment archive to the builder

### PR DESCRIPTION
**Pull request by iulianbarbu**
## Description of change

Send the deployment archive to the builder.

## How has this been tested? (if applicable)

N/A, will follow up soon.

### CODE REVIEW

File: deployer/src/deployment/queue.rs [hunk 1]
```diff
@@ -79,6 +88,17 @@ pub async fn task(
...
+                        match builder_client.build(BuildRequest {
+                            deployment_id: queued.id.to_string(),
+                            archive: queued.data.clone(),
+                        }).await {
+                            Ok(inner) =>  {
+                                let response = inner.into_inner();
+                                info!(deployment_id = %queued.id, "shuttle-builder finished building the deployment: image length is {} bytes, is_wasm flag is {} and there are {} secrets", response.image.len(), response.is_wasm, response.secrets.len());
+                            },
+                            Err(err) => error!(deployment_id = %queued.id, "shuttle-builder errored while building: {}", err)
+                        };
```
Comment (oddgrd) [inline deployer/src/deployment/queue.rs:100]: Should we spawn this so it doesn't block the regular build?

Comment (iulianbarbu) [reply deployer/src/deployment/queue.rs:100]: Yep, definitely. Will do.

File: deployer/src/deployment/queue.rs [hunk 2]
```diff
@@ -79,6 +88,17 @@ pub async fn task(
...
 
+                        match builder_client.build(BuildRequest {
+                            deployment_id: queued.id.to_string(),
+                            archive: queued.data.clone(),
+                        }).await {
+                            Ok(inner) =>  {
+                                let response = inner.into_inner();
+                                info!(deployment_id = %queued.id, "shuttle-builder finished building the deployment: image length is {} bytes, is_wasm flag is {} and there are {} secrets", response.image.len(), response.is_wasm, response.secrets.len());
+                            },
+                            Err(err) => error!(deployment_id = %queued.id, "shuttle-builder errored while building: {}", err)
```
Comment (jonaro00) [inline deployer/src/deployment/queue.rs:99]: Will these show up in Deployer logs to user? They don't need to yet.

Comment (oddgrd) [reply deployer/src/deployment/queue.rs:99]: Hmm, like it is currently I think they will. Perhaps if we instrument the span when we spawn this and don't include the deployment_id we will skip sending it in the deploy layer? Or rename the deployment_id.

**Review by jonaro00 — APPROVED** [review]

File: deployer/src/deployment/queue.rs (hunk 2 above)
Comment (iulianbarbu) [reply deployer/src/deployment/queue.rs]: I renamed the `deployment_id` for now. We can look further into it once it's on staging. In parallel I'll continue my local testing.

File: deployer/src/deployment/queue.rs (hunk 1 above)
Comment (iulianbarbu) [reply deployer/src/deployment/queue.rs]: Done.

File: deployer/src/deployment/queue.rs (hunk 2 above)
Comment (oddgrd) [reply deployer/src/deployment/queue.rs]: Ah, okay, I think this will work, but we'll need to see in testing.

**Review by oddgrd — APPROVED** [review]
LGTM, but I'm not sure if we need to merge it before we test it

**Comment by iulianbarbu** [comment]
> LGTM, but I'm not sure if we need to merge it before we test it

This PR is against my fork, so we can not deploy to staging from here. I could open a new PR though with a branch on the shuttle repo.

**Comment by oddgrd** [comment]
Oh, I didn't see, I think just merge it then :+1:

**Comment by iulianbarbu** [comment]
Closing this in favor of #1251

**Pull request closed without merging**